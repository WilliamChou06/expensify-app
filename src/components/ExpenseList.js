import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import selectExpenses from "../selectors/expenses";


// Problem with ternary operator here
export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-desktop">Expense</div>
            <div className="show-for-desktop">Amount</div>
        </div>
        <div className="list-body">
            {
                props.expenses.length < 1 ? (
                    <div className="list-item list-item--message">
                        <span>No expenses</span>
                    </div>
                ) : (
                    props.expenses.map((expense) => (
                        <ExpenseListItem key={expense.id} {...expense} />
                    ))
                )
            }
        </div>
    </div>
);

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseList);

