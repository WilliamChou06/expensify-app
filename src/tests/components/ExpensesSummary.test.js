import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("should render ExpensesSummary correctly", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={5012} />);
    expect(wrapper).toMatchSnapshot();
})

test("should render ExpensesSummary correctly", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={45099} />);
    expect(wrapper).toMatchSnapshot();
})