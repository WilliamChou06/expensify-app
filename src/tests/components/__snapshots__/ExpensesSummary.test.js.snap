// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $50.12
  </p>
</div>
`;

exports[`should render ExpensesSummary correctly 2`] = `
<div>
  <p>
    Viewing 
    3
     
    expenses
     totalling 
    $450.99
  </p>
</div>
`;
